lib/modules/manager/nix/extract.ts

```diff
@@ -72,81 +65,138 @@ export async function extractPackageFile(
       continue;
     }
 
+    // flakeLocked example: { rev: '56a49ffef2908dad1e9a8adef1f18802bc760962', type: 'github' }
     const flakeLocked = flakeInput.locked;
+    // flakeOriginal example: { owner: 'NuschtOS', repo: 'search', type: 'github' }
     const flakeOriginal = flakeInput.original;
 
-    // istanbul ignore if: if we are not in a root node then original and locked always exist which cannot be easily expressed in the type
-    if (flakeLocked === undefined || flakeOriginal === undefined) {
+    if (flakeLocked === undefined) {
+      logger.debug(
+        { flakeLockFile, flakeInput },
+        `input is missing locked, skipping`,
+      );
+      continue;
+    }
+
+    if (flakeOriginal === undefined) {
+      logger.debug(
+        { flakeLockFile, flakeInput },
+        `input is missing original, skipping`,
+      );
+      continue;
+    }
+
+    // indirect inputs cannot be reliably updated because they depend on the flake registry
+    if (flakeOriginal.type === 'indirect' || flakeLocked.type === 'indirect') {
+      logger.debug(
+        { flakeLockFile, flakeInput },
+        `input is type ${flakeOriginal.type}, skipping`,
```

Comment:
The log message is confusing here if `flakeLocked.type` is `indirect` but `flakeOriginal.type` is something else.